### PR TITLE
Devsol 642: Move client level configurations from the security settings to the template settings

### PR DESCRIPTION
**Pull request by sdamodaran**
This includes the code for the SDK classes for the TemplateAuth schemes at the model level.

### CODE REVIEW

File: Apigee/SmartDocs/TemplateAuth.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+use Apigee\SmartDocs\Security\TemplateAuthScheme;
+
+class TemplateAuth extends APIObject
+{
+
+    /**
+     * Initializes this object.
+     *
+     * @param \Apigee\Util\OrgConfig $config
+     *   The mangement endpoint config.
```
Comment (ossobuffo) [inline Apigee/SmartDocs/TemplateAuth.php]: Spelling error

File: Apigee/SmartDocs/TemplateAuth.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+     * @return array
+     *   An array of TemplateAuthScheme subclasses.
+     */
+    public function loadAllSchemes()
+    {
+        try {
+            $this->get();
+            $schemes = array();
+            foreach ($this->responseObj as $scheme_array) {
+                $scheme                      = TemplateAuthScheme::fromArray($scheme_array);
```
Comment (ossobuffo) [inline Apigee/SmartDocs/TemplateAuth.php]: All these extra spaces are unnecessary.

File: Apigee/SmartDocs/Security/TemplateAuthScheme.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+   * @param array $parameters
+   */
+  public function __construct(array $parameters)
+  {
+    foreach ($parameters as $key => $value) {
+      if(property_exists($this, $key)) {
+        $this->$key = $value;
+      }
+    }
+  }
```
Comment (ossobuffo) [inline Apigee/SmartDocs/Security/TemplateAuthScheme.php]: Indentation from lines 13 to 55 should be 4 spaces, not 2.

File: Apigee/SmartDocs/Security/ApiKeyTemplateAuthScheme.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    public function toArray($is_update = false)
+    {
+        $retVal = parent::toArray($is_update);
+        $retVal += array(
+            'keyName' => $this->keyName,
+            'keyValue' => $this->keyValue
+        );
+        return $retVal;
+    }
+}
```
Comment (ossobuffo) [inline Apigee/SmartDocs/Security/ApiKeyTemplateAuthScheme.php:82]: Add a carriage return here.

File: Apigee/ManagementAPI/Developer.php [hunk 5]
```diff
@@ -450,7 +450,7 @@ public function save($force_update = false, $old_email = null)
         // We must also do this when creating a developer ex nihilo in order
         // to set initial status. Otherwise new developer will have default
         // status, which is generally 'approved'.
-        $this->post($old_email . '?action=' . $cached_status);
+        $this->post($this->email . '?action=' . $cached_status);
```
Comment (ossobuffo) [inline Apigee/ManagementAPI/Developer.php:453]: Can you tell me why this change was made? It makes it impossible to change the developer's email.
EDIT: On second thought, maybe not. Is this an unrelated bug you discovered?

Comment (sdamodaran) [inline Apigee/ManagementAPI/Developer.php:453]: This was discovered when we had a request from NTT.com to have the developer email editable, which is tracked under DEVSOL-1013.

So when update the developer email $old_email is no more available and doing a POST to $old_email fails.

But if we do a POST to $this->email it will be the updated email.

This issue is tracked under DEVSOL-1291
